src/App.tsx

```diff
@@ -14,37 +15,107 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+  None = '',
+}
 
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState<string[]>([
+    ...goodsFromServer,
+  ]);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const sortGoods = (goods: string[], type: SortType): string[] => {
+    const sorted = [...goods];
+
+    switch (type) {
+      case SortType.Alphabetical:
+        sorted.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.Length:
+        sorted.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      sorted.reverse();
+    }
+
+    return sorted;
+  };
+
+  const handleSort = (type: SortType) => {
+    const newSortType = type || sortType;
+
+    setSortType(newSortType);
+
+    const sorted = sortGoods(goodsFromServer, newSortType);
+
+    setSortedGoods(sorted);
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+    setSortedGoods(prevGoods => [...prevGoods].reverse());
+  };
+
+  const resetSort = () => {
+    setSortType(SortType.None);
+    setIsReversed(false);
+    setSortedGoods([...goodsFromServer]);
+  };
+
+  const buttonStyle = (type: SortType) => (sortType === type ? '' : 'is-light');
```

Comment:
The `buttonStyle` function name does not clearly indicate what it does. Following the checklist item 4, function names should start with a verb to describe the action they perform. Consider renaming it to something like `getButtonStyle` to improve readability and code style.